std/sumtype.d

```diff
@@ -790,6 +790,18 @@ public:
             return this.match!hashOf;
         }
     }
+
+    /**
+     * Returns the index of the type of the `SumType`'s current value in the
+     * `SumType`'s `Types`.
+     *
+     * If the `SumType` is qualified, then its qualifiers are applied to
+     * `Types` before determining the index.
```

Comment:
This second paragraph should be removed.

Reply:
Should it? In the case of something like `const(SumType!(A, B))(const(A)())`, the type of the value is `const(A)`, which is not present in `Types`. The first paragraph alone is insufficient to determine the correct behavior in this case (although I'll grant that it is not difficult to guess).

Reply:
Well, you aren't applying the qualifier. You are just returning the tag.

The paragraph suggests something different happens when a qualifier is used.

Perhaps a rewrite is needed. Maybe:

```
If the SumType is qualified, the index indicates the Type if it were unqualified.
```

Reply:
The specific wording you suggest is incorrect, because the type in `Types` *might* already be the qualified version, but I agree that a rewrite is the best solution here.